src/main/java/com/sciamus/contractanalyzer/infrastructure/adapter/mongo/MongoReportPersistenceAdapter.java

```diff
@@ -83,7 +84,7 @@ public ReportInfrastructureDTO findById(String id) {
 
         Pageable pageable;
 
-        if (reportFilterParameters.sortingOrder != null && reportFilterParameters.sortingProperty != null && !reportFilterParameters.sortingOrder.isBlank() && !reportFilterParameters.sortingProperty.isBlank()) {
+        if (!(Strings.isNullOrEmpty(reportFilterParameters.sortingOrder) || Strings.isNullOrEmpty(reportFilterParameters.sortingProperty))) {
```

Comment:
## warn

z racji swojego skomplikowania, łatwo przegapić błąd: drugie wyrażenie nie ma wykrzyknika? Jak to ominąć? Otóż Nor z dwój negacji to And braku negacji :) :
!(!a||!b) = a && b

Jeszcze, patrząc na to, chyba w aktualnym rozwiązaniu powinien w środku być &&? To jest jeszcze o tyle ciekawe, że sortingOrder może być domyślny, czyli:

`var executiveSortingOrder = Option.of(reportFilterParameters.sortingOrder).getOrElse(DEFAULT_SORTING)`

czyli, u nas dużo łatwiej byłoby gdybyśmy jeszcze odwrócili:

```
if(nullOrEmpty(reportFilterParameters.sortingProperty)){
    pageableWithoutSorting
} else{
    pageableWithSorting
}
```

BTW wyciągnij tworzenie pageable do osobnej metodki (bez zmiennej pomocnicznej).

Reply:
hej,
imho źle spojrzałeś: 
!(Strings.isNullOrEmpty(reportFilterParameters.sortingOrder) || Strings.isNullOrEmpty(reportFilterParameters.sortingProperty) - negacja jest PRZED nawiasem, ergo jest to AND z dwóch negacji: !Strings.isNullOrEmpty(reportFilterParameters.sortingOrder) && !Strings.isNullOrEmpty(reportFilterParameters.sortingProperty

Reply:
nie, sorry, zwracam honor, masz rację, idea zamieniła mi tylko część, a nie całość, wyrażenia, teraz zwróciłem uwagę na nawias :/

Reply:
!(Strings.isNullOrEmpty(reportFilterParameters.sortingOrder || Strings.isNullOrEmpty(reportFilterParameters.sortingProperty) -> tak by było okej, prawda?

Reply:
jednak wracam do poprzedniego zdania :)

Reply:
jeszcze tutaj jedna rzecz wyszła - wymagamy, żeby były koniecznie 2 parametry sortujące, a możemy jeszcze umożliwić 2 zachowania: sortowanie po property z domyślnym orderem oraz sortowanie bez property z określonym orderem (muszę spojrzeć po czym wtedy sortuje, w sensie jakie jest domyślne property). Daj znać proszę czy i jakie zachowania mam dodać.

Reply:
tak, domyślność jest wskazana